files/stat.py

```diff
@@ -69,6 +69,179 @@
 - stat: path=/path/to/myhugefile get_md5=no
 '''
 
+RETURN = '''
+stat:
+    description: dictionary containing all the stat data
+    returned: success
+    type: dictionary
+    contains:
+        exists:
+            description: if the destination path actually exists or not
+            returned: success
+            type: boolean
+            sample: True
+        path:
+            description: The full path of the file/object to get the facts of
+            returned: success
```

Comment:
It looks like the path is returned only if 'success and path exists'. Look at the code for the case where the path doesn't exist. A dictionary with one element ('exists') is returned.